Minio/ApiEndpoints/ObjectOperations.cs

```diff
@@ -750,11 +782,61 @@ internal byte[] ReadFull(Stream data, int currentPartSize)
                 destObjectName = objectName;
             }
 
+            // Get Stats on the source object 
+            ObjectStat srcStats = await this.StatObjectAsync(bucketName, objectName, cancellationToken);
+
+            long srcByteRangeSize = 0L;
+
+            if (copyConditions != null)
+            {
+                srcByteRangeSize = copyConditions.GetByteRange();
+            }
+            long copySize = (srcByteRangeSize == 0) ? srcStats.Size : srcByteRangeSize;
+
+            if ((srcByteRangeSize > srcStats.Size) ||
+                ((srcByteRangeSize > 0) && (copyConditions.byteRangeEnd >= srcStats.Size)))
+                throw new ArgumentException("Specified byte range (" + copyConditions.byteRangeStart.ToString() + "-" + copyConditions.byteRangeEnd.ToString() + ") does not fit within source object (size=" + srcStats.Size.ToString() + ")");
+
+            if ((srcByteRangeSize > 0) && (srcByteRangeSize < Constants.MinimumPartSize))
+                throw new ArgumentException("Copy Source Range invalid if range < 5 MB");
```

Comment:
This check is not needed - it is valid to want to copy objects that are less than 5MB in size.